iceoryx_posh/include/iceoryx_posh/internal/mepoo/mepoo_segment.inl

```diff
@@ -76,8 +76,12 @@ inline SharedMemoryObjectType MePooSegment<SharedMemoryObjectType, MemoryManager
             .permissions(SEGMENT_PERMISSIONS)
             .create()
             .and_then([this](auto& sharedMemoryObject) {
-                this->setSegmentId(static_cast<uint64_t>(iox::rp::BaseRelativePointer::registerPtr(
-                    sharedMemoryObject.getBaseAddress(), sharedMemoryObject.getSizeInBytes())));
+                auto maybeSegmentId = iox::rp::BaseRelativePointer::registerPtr(sharedMemoryObject.getBaseAddress(),
+                                                                                sharedMemoryObject.getSizeInBytes());
+                if (maybeSegmentId.has_value())
+                {
+                    this->setSegmentId(static_cast<uint64_t>(maybeSegmentId.value()));
+                }
```

Comment:
@MatthiasKillat As the sad path is a fatal error, I would create a new error enum entry here. What do you think?

Reply:
@mossmaurice I would at least add a terminate, a todo and later call `IOX_PANIC` (for a fatal error). But we can also have an error code for it (INSUFFICIENT_SEGMENT_IDS or something like this).